python/ray/_private/worker.py

```diff
@@ -1905,9 +1905,24 @@ def shutdown(_exiting_interpreter: bool = False):
     _shutdown_all_compiled_dags()
 
     if _exiting_interpreter and global_worker.mode == SCRIPT_MODE:
-        # This is a duration to sleep before shutting down everything in order
-        # to make sure that log messages finish printing.
-        time.sleep(0.5)
+        # Checks whether all log messages for this job have been forwarded to stdout
```

Comment:
what if log_to_driver is false?

Reply:
In this case the queue will be always empty.